pkg/cloudprovider/cloudprovider.go

```diff
@@ -321,12 +322,21 @@ func (c *CloudProvider) instanceToMachine(ctx context.Context, ec2instance *ec2.
 	labels[v1alpha1.LabelInstanceAMIID] = aws.StringValue(ec2instance.ImageId)
 	labels[v1.LabelTopologyZone] = aws.StringValue(ec2instance.Placement.AvailabilityZone)
 	labels[v1alpha5.LabelCapacityType] = instance.GetCapacityType(ec2instance)
+	labels[v1.LabelOSStable] = instance.GetOS(ec2instance)
+
 	if tag, ok := lo.Find(ec2instance.Tags, func(t *ec2.Tag) bool { return aws.StringValue(t.Key) == v1alpha5.ProvisionerNameLabelKey }); ok {
 		labels[v1alpha5.ProvisionerNameLabelKey] = aws.StringValue(tag.Value)
 	}
 	if tag, ok := lo.Find(ec2instance.Tags, func(t *ec2.Tag) bool { return aws.StringValue(t.Key) == v1alpha5.ManagedByLabelKey }); ok {
 		labels[v1alpha5.ManagedByLabelKey] = aws.StringValue(tag.Value)
 	}
+	if tag, ok := lo.Find(ec2instance.Tags, func(t *ec2.Tag) bool { return aws.StringValue(t.Key) == v1alpha1.LabelWindowsVersion }); ok {
```

Comment:
Do we need to add this? I would have assumed that this would have been satisfied by L313 above that adds the label if there's only one value used?

Reply:
I don't like this either. However, there's some problem I haven't found the root cause of.

1. These tags is used by drift cycle. yes, it sounds unreasonable but it is!
2. In the drift cycle, the controller recovers all machine requirements from node labels.
3. In the current implementation, the LabelWindowsVersion and LabelWindowsVariant are very important. They must be presented.
4. There are two options to add these label
    1. Pass as kubelet arguments. I tried, but to my surprise, it doesn't work. I believe, there's some hidden issue in [karpenter core](https://github.com/aws/karpenter-core/blob/ab4eccca6536fa89c751953cad2c0611c5a9018c/pkg/controllers/provisioning/provisioner.go#L362) that the node labels are overwritten by karpenter core.
    2.  Pass as node tags and restore by karpenter. As current code